transcom/mymove
# INT B-20879 & I-13189 calculation fixes

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20879)
## [Issue ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=I-13189)

## Summary

There was a discrepancy between the description and AC of the feature so an issue was opened after demoing.

Per the issue wordage:
```
In the "Review SIT Extension Request" panel, the current day (today) is being used as part of the 
"Total Days Remaining" instead of "Total Days Used", "Total days in Origin SIT", and "Calculated total SIT days"

"Total days remaining" should go down by 1 and 
"Total days used", "Calculated total SIT days", and "Total days in Origin SIT" should go up by 1
```

This PR does the following:
- updates the `daysInSIT` private function in `pkg/services/sit_status/shipment_sit_status.go` to add one day ONLY when the SIT entry date is before today - this ensure that we are including today's date in the calculation by adding 1 day to that `if` statement
- removed subtracting of a day in the `sitFormatters` component when calculating the total days SIT used since we are changing data in the backend
- updated tests


### How to test

1. Follow instructions [in this PR](https://github.com/transcom/mymove/pull/13405)
2. Ensure dates are lined up right and calculated correct - including considering today in the `Total Days Used`, `Calculated total SIT days`, and `Total days in Origin/Destination SIT` calculations
3. Can use[ this calculator](https://www.timeanddate.com/date/duration.html) to check your work

## Screenshots
![Screenshot 2024-08-12 at 12 44 02 PM](https://github.com/user-attachments/assets/152be4da-0504-47f9-86c8-390b46c96ed7)
